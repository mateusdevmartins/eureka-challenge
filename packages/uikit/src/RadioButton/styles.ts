import styled from 'styled-components';

export const OutterView = styled.div`
  height: 12px;
  width: 12px;
  borderwidth: 2px;
  border: 2px solid;
  align-items: center;
  border-radius: 12px;
  display: flex;
  justify-content: center;
`;

export const InnerView = styled.div`
  height: 8px;
  width: 8px;
  border-radius: 8px;
`;
