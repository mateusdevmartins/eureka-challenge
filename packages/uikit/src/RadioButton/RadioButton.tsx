import { InnerView, OutterView } from './styles';

export interface RadioButtonProps {
  isSelected: boolean;
}

export function RadioButton({ isSelected }: RadioButtonProps) {
  return (
    <OutterView
      data-testid="outter-radio"
      style={{ borderColor: isSelected ? '#1C8EFF' : '#B5BAC8' }}
    >
      <InnerView
        data-testid="inner-radio"
        style={{
          backgroundColor: isSelected ? '#1C8EFF' : 'transparent',
        }}
      />
    </OutterView>
  );
}
