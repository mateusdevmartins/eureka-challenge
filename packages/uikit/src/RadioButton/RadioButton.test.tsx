import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { RadioButton } from './RadioButton';

describe('RadioButton should match snapshot', () => {
  const props1 = {
    isSelected: false,
  };
  const props2 = {
    isSelected: true,
  };
  describe('behavior', () => {
    const component = renderer.create(<RadioButton {...props1} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('behavior', () => {
    it('isSelected false', () => {
      const { getByTestId } = render(<RadioButton {...props1} />);

      expect(getByTestId('outter-radio').style.borderColor).toBe('#b5bac8');
      expect(getByTestId('inner-radio').style.backgroundColor).toBe(
        'transparent'
      );
    });

    it('isSelected true', () => {
      const { getByTestId } = render(<RadioButton {...props2} />);
      expect(getByTestId('outter-radio').style.borderColor).toBe('#1c8eff');
    });
  });
});
