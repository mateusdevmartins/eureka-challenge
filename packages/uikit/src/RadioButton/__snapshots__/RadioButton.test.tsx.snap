// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="styles__OutterView-pgnue4-0 emxFVC"
  data-testid="outter-radio"
  style={
    Object {
      "borderColor": "#B5BAC8",
    }
  }
>
  <div
    className="styles__InnerView-pgnue4-1 jmVbdE"
    data-testid="inner-radio"
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  />
</div>
`;
